### PR TITLE
feat(zeebe): extract connector mappings

### PR DESCRIPTION
**Pull request by marstamm**
Recording 2023-04-13 at 10 12 05

This PR adds an extractor for the `zeebe:TaskHeaders` used by connectors. This will add variable names from connector mappings to the overall scope.

We could add this extractor as an additional module, but I decided to implement it here, as we can re-use the feel utilities needed for parsing the result expression.

related to https://github.com/camunda/camunda-bpmn-js/issues/265

### CODE REVIEW

**Comment by nikku** [comment]
To better understand what this PR does: 

* Result variable defined (if defined) is added to global scope
* Result variables extracted (throught context) are added to global scope

Correct?

**Comment by marstamm** [comment]
Correct, except that they are not added to the global scope but to the nearest scoped parent. In most cases this will be the process scope, unless you use subprocesses

File: lib/zeebe/extractors/connectors.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        name: resultVariable.value
+      });
+    }
+
+    if (resultExpression) {
+
+      // parse with FEEL
+      const resultContext = getResultContext(resultExpression.value.substring(1));
+
+      const allVariables = toUnifiedFormat(resultContext.computedValue(), result);
```
Comment (nikku) [inline lib/zeebe/extractors/connectors.js:41]: Interesting name. :dancers:

File: test/spec/zeebe/ZeebeVariableResolver.spec.js [hunk 2]
```diff
@@ -633,6 +634,85 @@ describe('ZeebeVariableResolver', function() {
...
+    }));
+
+
+    it('should add variables from resultVariable header', inject(async function(variableResolver, elementRegistry) {
+
+      // given
+      const task = elementRegistry.get('resultVariable');
+
+      // when
+      const variables = await variableResolver.getVariablesForElement(task.businessObject);
```
Comment (nikku) [inline test/spec/zeebe/ZeebeVariableResolver.spec.js:671]: Great testing story!

File: test/spec/zeebe/ZeebeVariableResolver.spec.js [hunk 3]
```diff
@@ -633,6 +634,85 @@ describe('ZeebeVariableResolver', function() {
...
+
+      // given
+      const task = elementRegistry.get('both');
+
+      // when
+      const variables = await variableResolver.getVariablesForElement(task.businessObject);
+
+      // then
+      expect(variables).to.variableEqual([
+        { name: 'resultVariable', origin: [ 'both' ] },
```
Comment (nikku) [inline test/spec/zeebe/ZeebeVariableResolver.spec.js:707]: Both is the ID of the element, right?

**Review by nikku — APPROVED** [review]
Looks great. Readable, well tested :rocket:

File: test/spec/zeebe/ZeebeVariableResolver.spec.js (hunk 3 above)
Comment (marstamm) [reply test/spec/zeebe/ZeebeVariableResolver.spec.js:707]: Yes, that's the element ID 😄

Comment (nikku) [reply test/spec/zeebe/ZeebeVariableResolver.spec.js:707]: A little confusing. :)

Comment (marstamm) [reply test/spec/zeebe/ZeebeVariableResolver.spec.js:707]: I'll do some renameings on the things you mentioned before merging then 😄

File: lib/zeebe/util/feelUtility.js [hunk 4]
```diff
@@ -4,8 +4,8 @@ import {
 } from '@lezer/lr';
 
 import { EntriesContext } from './VariableContext';
-import { getExtensionElementsList } from './ExtensionElementsUtil';
-import { getParents } from './scopeUtil';
+import { getExtensionElementsList } from '../../base/util/ExtensionElementsUtil';
+import { getParents } from '../../base/util/scopeUtil';
```
[the remark this answers is not in the document]
Comment (marstamm) [reply lib/zeebe/util/feelUtility.js:8]: This actually gets all parents (not all scoped parents) 😄

[the remark this answers is not in the document]
Comment (nikku) [reply lib/zeebe/util/feelUtility.js:8]: HAHA, so much about naming. So my bad in that case. I was confused about `scopeUtil` exposing such a utility.

**Pull request merged**